core/src/main/java/com/google/instrumentation/stats/MutableView.java

```diff
@@ -80,22 +81,43 @@ ViewDescriptor getViewDescriptor() {
 
     @Override
     void record(Map<String, String> tags, double value) {
-      // TODO(songya): compare tags with tagKeys in ViewDescriptor
-      for (MutableDistribution distribution : mutableDistributions) {
-        distribution.add(value);
+      final List<TagKey> tagKeys = this.distributionViewDescriptor.getTagKeys();
```

Comment:
maybe note that either uniqueness in tag key isn't important, or it is enforced elsewhere.